library/core/src/panic/location.rs

```diff
@@ -44,6 +46,42 @@ pub struct Location<'a> {
     _filename: PhantomData<&'a str>,
 }
 
+#[stable(feature = "panic_hooks", since = "1.10.0")]
+impl PartialEq for Location<'_> {
+    fn eq(&self, other: &Self) -> bool {
+        self.file() == other.file() && self.line == other.line && self.col == other.col
+    }
+}
+
+#[stable(feature = "panic_hooks", since = "1.10.0")]
+impl Eq for Location<'_> {}
+
+#[stable(feature = "panic_hooks", since = "1.10.0")]
+impl Ord for Location<'_> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.file()
+            .cmp(other.file())
+            .then_with(|| self.line.cmp(&other.line))
+            .then_with(|| self.col.cmp(&other.col))
+    }
+}
+
+#[stable(feature = "panic_hooks", since = "1.10.0")]
+impl PartialOrd for Location<'_> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+#[stable(feature = "panic_hooks", since = "1.10.0")]
+impl Hash for Location<'_> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.file().hash(state);
+        self.line.hash(state);
+        self.col.hash(state);
+    }
+}
+
```

Comment:
Can you add some test coverage for these? That would help prevent future accidental regressions.

Reply:
`Location::internal_constructor` does just that :) and there is an existing test file at `library/coretests/tests/panic/location.rs`

Reply:
It appears that the function you're referencing no longer exists and was removed when the file was converted to a non-null pointer instead of a real string. Specifically here: b541f93

Reply:
Interesting; I guess it could be brought back as a function taking a `&CStr`.

Reply:
I've added tests now using some extra helper modules in separate files.